lib/topological_inventory/openshift/operations/core/service_catalog_client.rb

```diff
@@ -31,24 +30,22 @@ def sources_api_client
 
           def order_service_plan(plan_name, service_offering_name, additional_parameters)
             payload = build_payload(plan_name, service_offering_name, additional_parameters)
-            connection.create_service_instance(payload)
+            servicecatalog_connection.create_service_instance(payload)
           end
 
           def wait_for_provision_complete(name, namespace)
-            service_instance = nil
+            field_selector = "involvedObject.kind=ServiceInstance,involvedObject.name=#{name}"
 
-            loop do
-              sleep(sleep_poll)
+            watch = kubernetes_connection.watch_events(:namespace => namespace, :field_selector => field_selector)
+            watch.each do |notice|
+              event = notice.object
 
-              service_instance = connection.get_service_instance(name, namespace)
+              logger.info("#{event.involvedObject.name}: message [#{event.message}] reason [#{event.reason}]")
```

Comment:
@agrare Is this event going to get posted to Kafka to be accessible by the CatalogAPI?

Reply:
@mkanoor the task will be updated with the reason and message when it gets completed.  I could see updating the task while the provision is running, but we won't be posting the events to kafka

Reply:
Updating the task while waiting for it to complete is going to take some refactoring because those two are in two different classes, lets do that in a followup PR

Reply:
@agrare Here's the kind of the progress I was making before this PR on the task updating Madhu is requesting (https://github.com/ManageIQ/topological_inventory-openshift/compare/master...eclarizio:update_task_with_progress_events). Obviously, it will need to change a bit since we're not sleep-looping, but basically all I think we need to do is make an `update_task` call, then we can handle it on the catalog side.

The only problem here (I think) is that events are going to pile up so if it takes say 3 events to complete, the first time we're going to get 1, the second we're going to get 1 and 2, and the third we're going to get 1, 2, and 3, right? I guess it depends on how `watch_events` works but that was my understanding from my brief chat with Madhu.

Reply:
@eclarizio kubernetes will send all notices from before you started your watch for the entity (one at a time) until you catch up to current time, then seamlessly your thread blocks until new events are raise.

It doesn't send the full summary of all historical events every time any event is raised if that's what you mean?

Reply:
@agrare That confusion might have been because of the way the REST API call returns all the events every time you call it.

Reply:
@mkanoor ah yes that is what happens when you don't pass a resourceVersion in, the kubeclient call keeps track of the resourceVersion so you only get events since that version after each block yield